### DISCUSSION TITLE
Install mingw64 packages on fedora/ubuntu ?

### DISCUSSION THREAD

**Discussion by hbadi** [msys2/MINGW-packages#11962] [Q&A]
Hi,

Is there a way to install msys2/mingw64 packages on a Linux distro (Ubuntu or Fedora for instance) in order to use them in a cross-compilation process ?
I use the x86_64-w64-mingw32 toolchain compilation for Fedora and it'd be great to use some msys2-mingw64 native precompiled libraries which are known to be tricky to compile.

Thanks.

**Comment by Biswa96** [comment]
There is a related proposal here https://fedoraproject.org/wiki/Changes/F37MingwUCRT. Though I am not sure if it will be using packages from msys2 directly.

**Comment by hbadi** [comment]
Hello @Biswa96 ,
Thank you for the related link. I'd like to use the pacman package manager directly in Fedora...

**Reply by 1480c1** [reply]
That would be harder as the base cygwin base does not work properly under wine atm. You could try getting a static pacman running under Fedora and add msys2's repos to the /etc/pacman.conf file, but there's no guarantee of anything working that way either

**Comment by hbadi** [comment]
My goal is to compile for windows and test executable with windows.
I tried to modify /etc/pacman.conf and i added some files in /etc/pacmac.d/mirrorlist* (from msys2).
Then `pacman-key --init`
When i when to search a package `pacman -Ss packagename` i get this error :
```
error: mingw64: signature from "Christoph Reiter (MSYS2 development key) <reiter.christoph@gmail.com>" is unknown trust
error: ucrt64: signature from "Christoph Reiter (MSYS2 development key) <reiter.christoph@gmail.com>" is unknown trust
error: clang64: signature from "Christoph Reiter (MSYS2 development key) <reiter.christoph@gmail.com>" is unknown trust
error: database 'mingw64' is not valid (invalid or corrupted database (PGP signature))
error: database 'ucrt64' is not valid (invalid or corrupted database (PGP signature))
error: database 'clang64' is not valid (invalid or corrupted database (PGP signature))
```

**Comment by hbadi** [comment]
It works when i add `SigLevel = Optional TrustAll`

**Comment by wszqkzqk** [comment]
I've launched a related project to install and use packages from msys2 on Linux: https://github.com/wszqkzqk/LinSYS2

It may be helpful for you.
